### PR TITLE
Fix sonarCloud warnings at tools/manifest-templator/manifest-templator.go

### PR DESCRIPTION
**Pull request by nunnatsa**
1. Define a constant instead of duplicating this literal "rbac.authorization.k8s.io/v1" 4 times.
2. Refactor this method to reduce its Cognitive Complexity from 29 to the 15 allowed.

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: tools/manifest-templator/manifest-templator.go [hunk 1]
```diff
@@ -72,40 +77,299 @@ var (
...
+	writeOperatorCrd()
+
+	writev2vCrd()
+	writev2vVirtCrd()
+	writeOperatorYaml(deployments, services)
+
+	// Write out rbac
+	// since maps are not ordered we must enforce one before writing
+	writeSaYaml(serviceAccounts)
+	writeCrbYaml(roleBindings, clusterRoleBindings)
```
Comment (zcahana) [inline tools/manifest-templator/manifest-templator.go:199]: I'd suggest to rename the following methods while keeping the following principles:
- Consistently ~use~/not use the "Yaml" suffix
- Avoid unintuitive acronyms like "Crb"
- Use uppercase acronyms where possible (although adjacent uppercase acronyms are confusing too, like V2VCRD).

My suggestion:
```go
writeOperatorCR()
writeOperatorCRD()
writeV2vCR()                      // Actually here I'm not sure if writev2vCR(), or writeV2VCR()
writeV2vCRD()                     // ditto
writeOperatorDeployments()
writeOperatorServices()           // Split this from the "deloyments+services" function
writeServiceAccounts()
writeClusterRoleBindings()
writeClusterRoles()
```

File: tools/manifest-templator/manifest-templator.go [hunk 2]
```diff
@@ -72,40 +77,299 @@ var (
...
+		for _, clusterPermission := range strategySpec.ClusterPermissions {
+			serviceAccounts[clusterPermission.ServiceAccountName] = createServiceAccount(clusterPermission)
+			clusterPermissions = append(clusterPermissions, createClusterRole(clusterPermission))
+			clusterRoleBindings = append(clusterRoleBindings, createClusterRoleBinding(clusterPermission))
+		}
+	}
+
+	// Write out CRDs and CR
+	writeOperatorCr()
+	// Write out deployments
```
Comment (zcahana) [inline tools/manifest-templator/manifest-templator.go:189]: This comment is incorrectly placed.

File: tools/manifest-templator/manifest-templator.go [hunk 3]
```diff
@@ -40,6 +41,10 @@ import (
 	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
+const (
+	rbacAPI = rbacv1.GroupName + "/v1"
```
[the remark this answers is not in the document]
Comment (nunnatsa) [reply tools/manifest-templator/manifest-templator.go:45]: This can't be compiled.

[the remark this answers is not in the document]
Comment (nunnatsa) [reply tools/manifest-templator/manifest-templator.go:45]: Make it var and move the asigment to the `init` method

File: tools/manifest-templator/manifest-templator.go (hunk 2 above)
Comment (nunnatsa) [reply tools/manifest-templator/manifest-templator.go:189]: Done, except for the spliting the deployment and the services. This method is wtiting to the same file.

File: tools/manifest-templator/manifest-templator.go (hunk 1 above)
Comment (nunnatsa) [reply tools/manifest-templator/manifest-templator.go:199]: Done, except for the spliting the deployment and the services. This method is wtiting to the same file.

File: tools/manifest-templator/manifest-templator.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (zcahana) [reply tools/manifest-templator/manifest-templator.go]: Initializing directly in the `var( ... )` block is perfectly fine too, probably even better and clearer. Not too important though.

**Comment by zcahana** [comment]
Looks good!

/lgtm
/approve

**Pull request merged**